packages/core/src/common/types.ts

```diff
@@ -130,3 +130,5 @@ export namespace ArrayUtils {
 export function unreachable(_never: never, message: string = 'unhandled case'): never {
     throw new Error(message);
 }
+
+export type ResultPromise<T> = Promise<T | undefined | null>;
```

Comment:
@alvsan09 I would probably keep the same name `ProviderResult` unless you have a strong reason to deviate? I don't think `ResultPromise` would be a good name for this type.

Reply:
I don't think we benefit from this type at all at them moment, I would remove it.

Reply:
Specifically I don't think it's super useful to use for implementations, which is its main use right now.

Looking at VS Code they have a `ProviderResult` used in definitions, but they never mention it in implementations.